core/ibft.go

```diff
@@ -462,6 +462,10 @@ func (i *IBFT) handleRoundChangeMessage(view *proto.View, quorum uint64) *proto.
 		return nil
 	}
 
+	if !messages.HasUniqueSenders(msgs) {
+		return nil
+	}
```

Comment:
Why do you need this check, since the messaging system guarantees unique senders of top-level messages such as the ones you've requested here?

Reply:
Removed

[5c95b5b](https://github.com/madz-lab/go-ibft/pull/10/commits/5c95b5b0eacee1a4de7080edd3cf33ac30173ae3)